### ISSUE TITLE
Angular: many to many with user broken

### ISSUE THREAD

**Issue by pascalgrimaud** [jhipster/generator-jhipster#12928]
##### **Overview of the issue**

When updating an entity which has relationship many to many with user, it's broken

![Capture d’écran de 2020-11-01 17-59-45](https://user-images.githubusercontent.com/9156882/97809673-9a8f6080-1c6e-11eb-8a1b-149261adfe1e.png)

##### **Reproduce the error**

Detected by daily builds: https://github.com/hipster-labs/jhipster-daily-builds/actions?query=workflow%3A%22Angular+Maven%22

Not detected in current main, as it uses Cypress and we don't have test for many-to-many

##### **Related issues**

Following https://github.com/jhipster/generator-jhipster/pull/12889 by @kaidohallik 

##### **Suggest a Fix**

Add back the condition, with eslint disable
Otherwise, it will complain with:

```
/home/pgrimaud/tmp/08-ngx/src/main/webapp/app/entities/test-infinite-scroll/test-infinite-scroll-update.component.ts
  105:9  error  Unnecessary conditional, the types have no overlap  @typescript-eslint/no-unnecessary-condition
```

The condition would be:

```typescript
    // eslint-disable-next-line
    if (selectedVals !== undefined) {
      for (let i = 0; i < selectedVals.length; i++) {
        if (option.id === selectedVals[i].id) {
          return selectedVals[i];
        }
      }
    }
```

##### **JHipster Version(s)**

Current main, specially since https://github.com/jhipster/generator-jhipster/pull/12889

##### **JHipster Version(s)**

```
app-ng-2@0.0.1-SNAPSHOT /home/pgrimaud/tmp/08-ngx
└── generator-jhipster@6.10.4 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "baseName": "appNg2",
    "packageName": "io.github.jhipster.app",
    "packageFolder": "io/github/jhipster/app",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "buildTool": "maven",
    "rememberMeKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "useSass": false,
    "applicationType": "monolith",
    "testFrameworks": ["gatling", "protractor"],
    "jhiPrefix": "custom",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en"],
    "ci": true,
    "serviceDiscoveryType": false,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "skipUserManagement": false,
    "clientTheme": "none",
    "clientThemeVariant": "primary",
    "skipClient": false,
    "skipServer": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "reactive": false,
    "blueprints": [],
    "otherModules": [],
    "pages": [],
    "jhipsterVersion": "6.10.4",
    "creationTimestamp": 1604247041670,
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "entities": [
      "Operation",
      "FieldTestEntity",
      "FieldTestInfiniteScrollEntity",
      "FieldTestMapstructAndServiceClassEntity",
      "FieldTestPaginationEntity",
      "FieldTestServiceClassAndJpaFilteringEntity",
      "FieldTestServiceImplEntity",
      "TestCustomTableName",
      "TestEntity",
      "TestInfiniteScroll",
      "TestMapstruct",
      "TestPagination",
      "TestServiceClass",
      "TestServiceImpl",
      "TestTwoRelationshipsSameEntity",
      "TestManyRelPaginDTO",
      "TestManyToMany",
      "TestManyToOne",
      "TestOneToOne",
      "EntityWithDTO",
      "EntityWithServiceClassAndPagination",
      "EntityWithServiceImplAndPagination",
      "EntityWithServiceImplAndDTO",
      "EntityWithPaginationAndDTO",
      "EntityWithServiceClassPaginationAndDTO",
      "EntityWithServiceImplPaginationAndDTO",
      "Division",
      "Place",
      "SuperMegaLargeTestEntity",
      "MapsIdParentEntityWithoutDTO",
      "MapsIdChildEntityWithoutDTO",
      "MapsIdParentEntityWithDTO",
      "MapsIdChildEntityWithDTO",
      "MapsIdUserProfileWithDTO",
      "FieldTestEnumWithValue",
      "BankAccount",
      "Label",
      "JpaFilteringRelationship",
      "JpaFilteringOtherSide"
    ]
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Operation {
  date Instant required,
  description String,
  amount BigDecimal required
}
entity FieldTestEntity {
  stringTom String,
  stringRequiredTom String required,
  stringMinlengthTom String minlength(0),
  stringMaxlengthTom String maxlength(20),
  stringPatternTom String pattern(/^[a-zA-Z0-9]*$/),
  numberPatternTom String pattern(/^[0-9]+$/),
  numberPatternRequiredTom String pattern(/^[0-9]+$/) required,
  integerTom Integer,
  integerRequiredTom Integer required,
  integerMinTom Integer min(0),
  integerMaxTom Integer max(100),
  longTom Long,
  longRequiredTom Long required,
  longMinTom Long min(0),
  longMaxTom Long max(100),
  floatTom Float,
  floatRequiredTom Float required,
  floatMinTom Float min(0),
  floatMaxTom Float max(100),
  doubleRequiredTom Double required,
  doubleMinTom Double min(0),
  doubleMaxTom Double max(100),
  bigDecimalRequiredTom BigDecimal required,
  bigDecimalMinTom BigDecimal min(0),
  bigDecimalMaxTom BigDecimal max(100),
  localDateTom LocalDate,
  localDateRequiredTom LocalDate required,
  instantTom Instant,
  instantRequiredTom Instant required,
  zonedDateTimeTom ZonedDateTime,
  zonedDateTimeRequiredTom ZonedDateTime required,
  durationTom Duration,
  durationRequiredTom Duration required,
  booleanTom Boolean,
  booleanRequiredTom Boolean required,
  enumTom EnumFieldClass,
  enumRequiredTom EnumRequiredFieldClass required,
  uuidTom UUID,
  uuidRequiredTom UUID required,
  byteImageTom ImageBlob,
  byteImageRequiredTom ImageBlob required,
  byteImageMinbytesTom ImageBlob minbytes(0),
  byteImageMaxbytesTom ImageBlob maxbytes(10000),
  byteAnyTom AnyBlob,
  byteAnyRequiredTom AnyBlob required,
  byteAnyMinbytesTom AnyBlob minbytes(0),
  byteAnyMaxbytesTom AnyBlob maxbytes(10000),
  byteTextTom TextBlob,
  byteTextRequiredTom TextBlob required
}
entity FieldTestInfiniteScrollEntity {
  stringHugo String,
  stringRequiredHugo String required,
  stringMinlengthHugo String minlength(0),
  stringMaxlengthHugo String maxlength(20),
  stringPatternHugo String pattern(/^[a-zA-Z0-9]*$/),
  integerHugo Integer,
  integerRequiredHugo Integer required,
  integerMinHugo Integer min(0),
  integerMaxHugo Integer max(100),
  longHugo Long,
  longRequiredHugo Long required,
  longMinHugo Long min(0),
  longMaxHugo Long max(100),
  floatHugo Float,
  floatRequiredHugo Float required,
  floatMinHugo Float min(0),
  floatMaxHugo Float max(100),
  doubleRequiredHugo Double required,
  doubleMinHugo Double min(0),
  doubleMaxHugo Double max(100),
  bigDecimalRequiredHugo BigDecimal required,
  bigDecimalMinHugo BigDecimal min(0),
  bigDecimalMaxHugo BigDecimal max(100),
  localDateHugo LocalDate,
  localDateRequiredHugo LocalDate required,
  instantHugo Instant,
  instanteRequiredHugo Instant required,
  zonedDateTimeHugo ZonedDateTime,
  zonedDateTimeRequiredHugo ZonedDateTime required,
  durationHugo Duration,
  durationRequiredHugo Duration required,
  booleanHugo Boolean,
  booleanRequiredHugo Boolean required,
  enumHugo EnumFieldClass,
  enumRequiredHugo EnumRequiredFieldClass required,
  uuidHugo UUID,
  uuidRequiredHugo UUID required,
  byteImageHugo ImageBlob,
  byteImageRequiredHugo ImageBlob required,
  byteImageMinbytesHugo ImageBlob minbytes(0),
  byteImageMaxbytesHugo ImageBlob maxbytes(10000),
  byteAnyHugo AnyBlob,
  byteAnyRequiredHugo AnyBlob required,
  byteAnyMinbytesHugo AnyBlob minbytes(0),
  byteAnyMaxbytesHugo AnyBlob maxbytes(10000),
  byteTextHugo TextBlob,
  byteTextRequiredHugo TextBlob required
}
entity FieldTestMapstructAndServiceClassEntity {
  stringEva String,
  stringRequiredEva String required,
  stringMinlengthEva String minlength(0),
  stringMaxlengthEva String maxlength(20),
  stringPatternEva String pattern(/^[a-zA-Z0-9]*$/),
  integerEva Integer,
  integerRequiredEva Integer required,
  integerMinEva Integer min(0),
  integerMaxEva Integer max(100),
  longEva Long,
  longRequiredEva Long required,
  longMinEva Long min(0),
  longMaxEva Long max(100),
  floatEva Float,
  floatRequiredEva Float required,
  floatMinEva Float min(0),
  floatMaxEva Float max(100),
  doubleRequiredEva Double required,
  doubleMinEva Double min(0),
  doubleMaxEva Double max(100),
  bigDecimalRequiredEva BigDecimal required,
  bigDecimalMinEva BigDecimal min(0),
  bigDecimalMaxEva BigDecimal max(100),
  localDateEva LocalDate,
  localDateRequiredEva LocalDate required,
  instantEva Instant,
  instanteRequiredEva Instant required,
  zonedDateTimeEva ZonedDateTime,
  zonedDateTimeRequiredEva ZonedDateTime required,
  durationEva Duration,
  durationRequiredEva Duration required,
  booleanEva Boolean,
  booleanRequiredEva Boolean required,
  enumEva EnumFieldClass,
  enumRequiredEva EnumRequiredFieldClass required,
  uuidEva UUID,
  uuidRequiredEva UUID required,
  byteImageEva ImageBlob,
  byteImageRequiredEva ImageBlob required,
  byteImageMinbytesEva ImageBlob minbytes(0),
  byteImageMaxbytesEva ImageBlob maxbytes(10000),
  byteAnyEva AnyBlob,
  byteAnyRequiredEva AnyBlob required,
  byteAnyMinbytesEva AnyBlob minbytes(0),
  byteAnyMaxbytesEva AnyBlob maxbytes(10000),
  byteTextEva TextBlob,
  byteTextRequiredEva TextBlob required
}
entity FieldTestPaginationEntity {
  stringAlice String,
  stringRequiredAlice String required,
  stringMinlengthAlice String minlength(0),
  stringMaxlengthAlice String maxlength(20),
  stringPatternAlice String pattern(/^[a-zA-Z0-9]*$/),
  integerAlice Integer,
  integerRequiredAlice Integer required,
  integerMinAlice Integer min(0),
  integerMaxAlice Integer max(100),
  longAlice Long,
  longRequiredAlice Long required,
  longMinAlice Long min(0),
  longMaxAlice Long max(100),
  floatAlice Float,
  floatRequiredAlice Float required,
  floatMinAlice Float min(0),
  floatMaxAlice Float max(100),
  doubleRequiredAlice Double required,
  doubleMinAlice Double min(0),
  doubleMaxAlice Double max(100),
  bigDecimalRequiredAlice BigDecimal required,
  bigDecimalMinAlice BigDecimal min(0),
  bigDecimalMaxAlice BigDecimal max(100),
  localDateAlice LocalDate,
  localDateRequiredAlice LocalDate required,
  instantAlice Instant,
  instanteRequiredAlice Instant required,
  zonedDateTimeAlice ZonedDateTime,
  zonedDateTimeRequiredAlice ZonedDateTime required,
  durationAlice Duration,
  durationRequiredAlice Duration required,
  booleanAlice Boolean,
  booleanRequiredAlice Boolean required,
  enumAlice EnumFieldClass,
  enumRequiredAlice EnumRequiredFieldClass required,
  uuidAlice UUID,
  uuidRequiredAlice UUID required,
  byteImageAlice ImageBlob,
  byteImageRequiredAlice ImageBlob required,
  byteImageMinbytesAlice ImageBlob minbytes(0),
  byteImageMaxbytesAlice ImageBlob maxbytes(10000),
  byteAnyAlice AnyBlob,
  byteAnyRequiredAlice AnyBlob required,
  byteAnyMinbytesAlice AnyBlob minbytes(0),
  byteAnyMaxbytesAlice AnyBlob maxbytes(10000),
  byteTextAlice TextBlob,
  byteTextRequiredAlice TextBlob required
}
entity FieldTestServiceClassAndJpaFilteringEntity {
  stringBob String,
  stringRequiredBob String required,
  stringMinlengthBob String minlength(0),
  stringMaxlengthBob String maxlength(20),
  stringPatternBob String pattern(/^[a-zA-Z0-9]*$/),
  integerBob Integer,
  integerRequiredBob Integer required,
  integerMinBob Integer min(0),
  integerMaxBob Integer max(100),
  longBob Long,
  longRequiredBob Long required,
  longMinBob Long min(0),
  longMaxBob Long max(100),
  floatBob Float,
  floatRequiredBob Float required,
  floatMinBob Float min(0),
  floatMaxBob Float max(100),
  doubleRequiredBob Double required,
  doubleMinBob Double min(0),
  doubleMaxBob Double max(100),
  bigDecimalRequiredBob BigDecimal required,
  bigDecimalMinBob BigDecimal min(0),
  bigDecimalMaxBob BigDecimal max(100),
  localDateBob LocalDate,
  localDateRequiredBob LocalDate required,
  instantBob Instant,
  instanteRequiredBob Instant required,
  zonedDateTimeBob ZonedDateTime,
  zonedDateTimeRequiredBob ZonedDateTime required,
  durationBob Duration,
  durationRequiredBob Duration required,
  booleanBob Boolean,
  booleanRequiredBob Boolean required,
  enumBob EnumFieldClass,
  enumRequiredBob EnumRequiredFieldClass required,
  uuidBob UUID,
  uuidRequiredBob UUID required,
  byteImageBob ImageBlob,
  byteImageRequiredBob ImageBlob required,
  byteImageMinbytesBob ImageBlob minbytes(0),
  byteImageMaxbytesBob ImageBlob maxbytes(10000),
  byteAnyBob AnyBlob,
  byteAnyRequiredBob AnyBlob required,
  byteAnyMinbytesBob AnyBlob minbytes(0),
  byteAnyMaxbytesBob AnyBlob maxbytes(10000),
  byteTextBob TextBlob,
  byteTextRequiredBob TextBlob required
}
entity FieldTestServiceImplEntity {
  stringMika String,
  stringRequiredMika String required,
  stringMinlengthMika String minlength(0),
  stringMaxlengthMika String maxlength(20),
  stringPatternMika String pattern(/^[a-zA-Z0-9]*$/),
  integerMika Integer,
  integerRequiredMika Integer required,
  integerMinMika Integer min(0),
  integerMaxMika Integer max(100),
  longMika Long,
  longRequiredMika Long required,
  longMinMika Long min(0),
  longMaxMika Long max(100),
  floatMika Float,
  floatRequiredMika Float required,
  floatMinMika Float min(0),
  floatMaxMika Float max(100),
  doubleRequiredMika Double required,
  doubleMinMika Double min(0),
  doubleMaxMika Double max(100),
  bigDecimalRequiredMika BigDecimal required,
  bigDecimalMinMika BigDecimal min(0),
  bigDecimalMaxMika BigDecimal max(100),
  localDateMika LocalDate,
  localDateRequiredMika LocalDate required,
  instantMika Instant,
  instanteRequiredMika Instant required,
  zonedDateTimeMika ZonedDateTime,
  zonedDateTimeRequiredMika ZonedDateTime required,
  durationMika Duration,
  durationRequiredMika Duration required,
  booleanMika Boolean,
  booleanRequiredMika Boolean required,
  enumMika EnumFieldClass,
  enumRequiredMika EnumRequiredFieldClass required,
  uuidMika UUID,
  uuidRequiredMika UUID required,
  byteImageMika ImageBlob,
  byteImageRequiredMika ImageBlob required,
  byteImageMinbytesMika ImageBlob minbytes(0),
  byteImageMaxbytesMika ImageBlob maxbytes(10000),
  byteAnyMika AnyBlob,
  byteAnyRequiredMika AnyBlob required,
  byteAnyMinbytesMika AnyBlob minbytes(0),
  byteAnyMaxbytesMika AnyBlob maxbytes(10000),
  byteTextMika TextBlob,
  byteTextRequiredMika TextBlob required
}
entity TestCustomTableName (test_custom_table_name_entity)
entity TestEntity
entity TestInfiniteScroll
entity TestMapstruct
entity TestPagination
entity TestServiceClass
entity TestServiceImpl
entity TestTwoRelationshipsSameEntity (test_multiple_rel)
entity TestManyRelPaginDTO (test_many_many_pagination_dto)
entity TestManyToMany
entity TestManyToOne
entity TestOneToOne
entity EntityWithDTO {
  emma String
}
entity EntityWithServiceClassAndPagination {
  enzo String
}
entity EntityWithServiceImplAndPagination {
  hugo String
}
entity EntityWithServiceImplAndDTO {
  louis String
}
entity EntityWithPaginationAndDTO {
  lea String
}
entity EntityWithServiceClassPaginationAndDTO {
  lena String
}
entity EntityWithServiceImplPaginationAndDTO {
  theo String
}
entity Division {
  name String required unique,
  shortName String unique,
  numberOfPeople Long,
  divisionType DivisionType required,
  colorBackground String,
  colorText String
}
entity Place {
  name String required,
  numberOfSeats Long,
  shortName String,
  colorBackground String,
  colorText String,
  description TextBlob
}
entity SuperMegaLargeTestEntity
entity MapsIdParentEntityWithoutDTO (parent_entity_wo_dto) {
  name String
}
entity MapsIdChildEntityWithoutDTO (child_entity_wo_dto) {
  date Instant
}
entity MapsIdParentEntityWithDTO (maps_id_parent_entity_withdto) {
  name String
}
entity MapsIdChildEntityWithDTO (maps_id_child_entity_withdto) {
  date Instant
}
entity MapsIdUserProfileWithDTO (maps_id_user_profile_withdto) {
  dateOfBirth Instant
}
entity FieldTestEnumWithValue (entity_with_enums) {
  myFieldA MyEnumA,
  myFieldB MyEnumB,
  myFieldC MyEnumC
}
entity BankAccount {
  name String required,
  guid UUID,
  bankNumber Integer,
  agencyNumber Long,
  lastOperationDuration Float,
  meanOperationDuration Double,
  meanQueueDuration Duration,
  balance BigDecimal required,
  openingDay LocalDate,
  lastOperationDate Instant,
  active Boolean,
  accountType BankAccountType,
  attachment AnyBlob,
  description TextBlob
}
entity Label {
  labelName String required minlength(3)
}
entity JpaFilteringRelationship {
  createdBy String,
  createdOn Instant
}
entity JpaFilteringOtherSide {
  createdBy String,
  createdOn Instant
}
enum EnumFieldClass {
  ENUM_VALUE_1,
  ENUM_VALUE_2,
  ENUM_VALUE_3
}
enum EnumRequiredFieldClass {
  ENUM_VALUE_1,
  ENUM_VALUE_2,
  ENUM_VALUE_3
}
enum DivisionType {
  SCHOOL,
  CLASS,
  SUBGROUP
}
enum MyEnumA {
  AAA,
  BBB
}
enum MyEnumB {
  AAA (aaa_aaa),
  BBB
}
enum MyEnumC {
  AAA (aaa_aaa),
  BBB (b and b)
}
enum BankAccountType {
  CHECKING,
  SAVINGS,
  LOAN
}

relationship OneToOne {
  TestCustomTableName{userOneToOne(login)} to User,
  TestEntity{userOneToOne(login)} to User,
  TestInfiniteScroll{userOneToOne(login)} to User,
  TestMapstruct{userOneToOne(login)} to User,
  TestPagination{userOneToOne(login)} to User,
  TestServiceClass{userOneToOne(login)} to User,
  TestServiceImpl{userOneToOne(login)} to User,
  TestTwoRelationshipsSameEntity{userOne} to User,
  TestTwoRelationshipsSameEntity{userTwo} to User,
  TestOneToOne{testEntity} to TestEntity{testOneToOne},
  TestOneToOne{testMapstruct} to TestMapstruct{testOneToOne},
  TestOneToOne{testServiceClass} to TestServiceClass{testOneToOne},
  TestOneToOne{testServiceImpl} to TestServiceImpl{testOneToOne},
  TestOneToOne{testInfiniteScroll} to TestInfiniteScroll{testOneToOne},
  TestOneToOne{testPagination} to TestPagination{testOneToOne},
  TestOneToOne{testCustomTableName} to TestCustomTableName{testOneToOne},
  TestOneToOne{superMegaLargeTestEntity} to SuperMegaLargeTestEntity{superMegaLargeTestOneToOne},
  SuperMegaLargeTestEntity{superMegaLargeUserOneToOne(login)} to User,
  MapsIdChildEntityWithoutDTO{mapsIdParentEntityWithoutDTO} to MapsIdParentEntityWithoutDTO{mapsIdChildEntityWithoutDTO} with jpaDerivedIdentifier,
  MapsIdChildEntityWithDTO{mapsIdParentEntityWithDTO} to MapsIdParentEntityWithDTO{mapsIdChildEntityWithDTO} with jpaDerivedIdentifier,
  MapsIdUserProfileWithDTO{user(login)} to User with jpaDerivedIdentifier,
  JpaFilteringRelationship{requiredOneToOne required} to JpaFilteringOtherSide{requiredOneToOneOtherSide},
  JpaFilteringRelationship{oneToOne} to JpaFilteringOtherSide{oneToOneOtherSide}
}
relationship OneToMany {
  BankAccount{operation} to Operation{bankAccount(name)},
  TestEntity{testCustomTableName} to TestCustomTableName{testEntity required},
  SuperMegaLargeTestEntity{superMegaLargeTestCustomTableName} to TestCustomTableName{superMegaLargeTestEntity required},
  TestEntity{testManyToOne} to TestManyToOne{testEntity},
  TestMapstruct{testManyToOne} to TestManyToOne{testMapstruct},
  TestServiceClass{testManyToOne} to TestManyToOne{testServiceClass},
  TestServiceImpl{testManyToOne} to TestManyToOne{testServiceImpl},
  TestInfiniteScroll{testManyToOne} to TestManyToOne{testInfiniteScroll},
  TestPagination{testManyToOne} to TestManyToOne{testPagination},
  TestCustomTableName{testManyToOne} to TestManyToOne{testCustomTableName},
  SuperMegaLargeTestEntity{superMegaLargeTestManyToOne} to TestManyToOne{superMegaLargeTestEntity},
  Division{divisionsPlace} to Place{owner(name)},
  JpaFilteringRelationship{requiredOneToMany required} to JpaFilteringOtherSide{requiredOneToManyOtherSide},
  JpaFilteringRelationship{oneToMany} to JpaFilteringOtherSide{oneToManyOtherSide}
}
relationship ManyToOne {
  TestCustomTableName{userOneToMany(login)} to User,
  TestEntity{userOneToMany(login) required} to User,
  TestInfiniteScroll{userOneToMany(login)} to User,
  TestMapstruct{userOneToMany(login)} to User,
  TestPagination{userOneToMany(login)} to User,
  TestServiceClass{userOneToMany(login)} to User,
  TestServiceImpl{userOneToMany(login)} to User,
  TestTwoRelationshipsSameEntity{firstRelationship} to TestEntity,
  TestTwoRelationshipsSameEntity{secondRelationship} to TestEntity,
  TestTwoRelationshipsSameEntity{firstUniqueRequiredRelation required} to Division,
  TestTwoRelationshipsSameEntity{secondUniqueRequiredRelation required} to Division,
  SuperMegaLargeTestEntity{superMegaLargeUserOneToMany(login) required} to User,
  BankAccount{user(login)} to User,
  JpaFilteringRelationship{requiredManyToOne required} to JpaFilteringOtherSide,
  JpaFilteringRelationship{manyToOne} to JpaFilteringOtherSide
}
relationship ManyToMany {
  Operation{label(labelName)} to Label{operation},
  TestCustomTableName{userManyToMany(login)} to User,
  TestEntity{userManyToMany(login)} to User,
  TestInfiniteScroll{userManyToMany(login)} to User,
  TestMapstruct{userManyToMany(login)} to User,
  TestPagination{userManyToMany(login)} to User,
  TestServiceClass{userManyToMany(login)} to User,
  TestServiceImpl{userManyToMany(login)} to User,
  TestManyRelPaginDTO{testMapstruct} to TestMapstruct{testManyRelPaginDTO},
  TestManyToMany{testEntity} to TestEntity{testManyToMany},
  TestManyToMany{testMapstruct} to TestMapstruct{testManyToMany},
  TestManyToMany{testServiceClass} to TestServiceClass{testManyToMany},
  TestManyToMany{testServiceImpl} to TestServiceImpl{testManyToMany},
  TestManyToMany{testInfiniteScroll} to TestInfiniteScroll{testManyToMany},
  TestManyToMany{testPagination} to TestPagination{testManyToMany},
  TestManyToMany{testCustomTableName} to TestCustomTableName{testManyToMany},
  TestManyToMany{superMegaLargeTestEntity} to SuperMegaLargeTestEntity{superMegaLargeTestManyToMany},
  Place{preferredDivision(name)} to Division{preferredPlace},
  SuperMegaLargeTestEntity{superMegaLargeUserManyToMany(login)} to User,
  JpaFilteringRelationship{requiredManyToMany required} to JpaFilteringOtherSide{requiredManyToManyOtherSide},
  JpaFilteringRelationship{manyToMany} to JpaFilteringOtherSide{manyToManyOtherSide}
}

paginate Operation, FieldTestInfiniteScrollEntity, TestInfiniteScroll with infinite-scroll
paginate FieldTestPaginationEntity, TestPagination, TestManyRelPaginDTO, EntityWithServiceClassAndPagination, EntityWithServiceImplAndPagination, EntityWithPaginationAndDTO, EntityWithServiceClassPaginationAndDTO, EntityWithServiceImplPaginationAndDTO, Label, JpaFilteringRelationship, JpaFilteringOtherSide with pagination
clientRootFolder Operation, Division, Place, BankAccount, Label with test-root
dto FieldTestMapstructAndServiceClassEntity, TestMapstruct, TestManyRelPaginDTO, EntityWithDTO, EntityWithServiceImplAndDTO, EntityWithPaginationAndDTO, EntityWithServiceClassPaginationAndDTO, EntityWithServiceImplPaginationAndDTO, MapsIdParentEntityWithDTO, MapsIdChildEntityWithDTO, MapsIdUserProfileWithDTO, BankAccount, JpaFilteringRelationship, JpaFilteringOtherSide with mapstruct
service FieldTestMapstructAndServiceClassEntity, FieldTestServiceClassAndJpaFilteringEntity, TestServiceClass, EntityWithServiceClassAndPagination, EntityWithServiceClassPaginationAndDTO, Label, JpaFilteringRelationship, JpaFilteringOtherSide with serviceClass
service FieldTestServiceImplEntity, TestServiceImpl, TestManyRelPaginDTO, EntityWithServiceImplAndPagination, EntityWithServiceImplAndDTO, EntityWithServiceImplPaginationAndDTO, MapsIdParentEntityWithDTO, MapsIdChildEntityWithDTO, MapsIdUserProfileWithDTO, BankAccount with serviceImpl
filter FieldTestServiceClassAndJpaFilteringEntity, TestServiceClass, TestServiceImpl, BankAccount, JpaFilteringRelationship, JpaFilteringOtherSide
angularSuffix TestEntity, SuperMegaLargeTestEntity with mySuffixAlt
angularSuffix TestTwoRelationshipsSameEntity, TestManyRelPaginDTO, TestManyToMany, TestManyToOne, TestOneToOne, BankAccount with mySuffix
readOnly Label

</pre>
</details>

##### **Environment and Tools**

openjdk version "11.0.9" 2020-10-20
OpenJDK Runtime Environment (build 11.0.9+11-Ubuntu-0ubuntu1.18.04.1)
OpenJDK 64-Bit Server VM (build 11.0.9+11-Ubuntu-0ubuntu1.18.04.1, mixed mode, sharing)

git version 2.29.0

node: v12.19.0

npm: 6.14.8

yeoman: 3.1.1

yarn: 1.22.5

Docker version 19.03.8, build afacb8b7f0

docker-compose version 1.27.3, build 4092ae5d

identical .jhipster/BankAccount.json
identical .jhipster/Division.json
identical .jhipster/EntityWithDTO.json
identical .jhipster/EntityWithPaginationAndDTO.json
identical .jhipster/EntityWithServiceClassAndPagination.json
identical .jhipster/EntityWithServiceClassPaginationAndDTO.json
identical .jhipster/EntityWithServiceImplAndDTO.json
identical .jhipster/EntityWithServiceImplAndPagination.json
identical .jhipster/EntityWithServiceImplPaginationAndDTO.json
identical .jhipster/FieldTestEntity.json
identical .jhipster/FieldTestEnumWithValue.json
identical .jhipster/FieldTestInfiniteScrollEntity.json
identical .jhipster/FieldTestMapstructAndServiceClassEntity.json
identical .jhipster/FieldTestPaginationEntity.json
identical .jhipster/FieldTestServiceClassAndJpaFilteringEntity.json
identical .jhipster/FieldTestServiceImplEntity.json
identical .jhipster/JpaFilteringOtherSide.json
identical .jhipster/JpaFilteringRelationship.json
identical .jhipster/Label.json
identical .jhipster/MapsIdChildEntityWithDTO.json
identical .jhipster/MapsIdChildEntityWithoutDTO.json
identical .jhipster/MapsIdParentEntityWithDTO.json
identical .jhipster/MapsIdParentEntityWithoutDTO.json
identical .jhipster/MapsIdUserProfileWithDTO.json
identical .jhipster/Operation.json
identical .jhipster/Place.json
identical .jhipster/SuperMegaLargeTestEntity.json
identical .jhipster/TestCustomTableName.json
identical .jhipster/TestEntity.json
identical .jhipster/TestInfiniteScroll.json
identical .jhipster/TestManyRelPaginDTO.json
identical .jhipster/TestManyToMany.json
identical .jhipster/TestManyToOne.json
identical .jhipster/TestMapstruct.json
identical .jhipster/TestOneToOne.json
identical .jhipster/TestPagination.json
identical .jhipster/TestServiceClass.json
identical .jhipster/TestServiceImpl.json
identical .jhipster/TestTwoRelationshipsSameEntity.json
INFO! Congratulations, JHipster execution is complete!

**Comment by mshima** [comment]
We should try to enable at cypress then.

**Comment by pascalgrimaud** [comment]
agree, Cypress should be at same level than Protractor
That's one of the reason why I'm keeping Protractor in Daily Builds for now

**Comment by pascalgrimaud** [comment]
cc @nassimerrahoui @adilabed @avdev4j as you worked a lot on Cypress

**Comment by pascalgrimaud** [comment]
More investigation on my side @mshima 
We were just lucky to find this issue with Protractor.

The main problem is simply because this function is not covered by the unit tests:
![Capture d’écran de 2020-11-01 19-07-17](https://user-images.githubusercontent.com/9156882/97810627-974ba300-1c75-11eb-89cb-213327114996.png)

**Comment by DanielFran** [comment]
@kaidohallik Can you review this issue?

**Comment by kaidohallik** [comment]
I'm sending fix PR shortly

**Comment by kaidohallik** [comment]
In `getSelected` signature was defined that `selectedVals` is never `undefined` and I remember that while doing PR I tested different cases here and this was true. But now I was able to reproduce in every many-to-many relationship that this can be `undefined` and this PR fixes that.  
I marked that `selectedVals` can be `undefined` and changed parameters order because optional parameter is not allowed before required parameter.
